core/src/main/java/kafka/log/remote/quota/RLMQuotaManager.java

```diff
@@ -77,16 +78,17 @@ public void updateQuota(Quota newQuota) {
         }
     }
 
-    public boolean isQuotaExceeded() {
+    public QuotaCheckResult checkQuota() {
         Sensor sensorInstance = sensor();
         try {
             sensorInstance.checkQuotas();
         } catch (QuotaViolationException qve) {
             LOGGER.debug("Quota violated for sensor ({}), metric: ({}), metric-value: ({}), bound: ({})",
                 sensorInstance.name(), qve.metric().metricName(), qve.value(), qve.bound());
-            return true;
+            long throttleTimeMs = QuotaUtils.throttleTime(qve, time.milliseconds());
+            return new QuotaCheckResult(true, throttleTimeMs);
```

Comment:
We can return long and remove returning a `QuotaCHeckResult` wrapper. This will avoid creating a wrapper instance and add the respective util check to determine the quota is exceeded if the value > 0.

Reply:
Makes sense. Made the changes as requested.